### PR TITLE
Cleanup for Editor Play Mode

### PR DESCRIPTION
**Pull request by mark-beiline**
Even if we don't support editor mode atm we can still expect the user to enter Play Mode for testing purposes.

Entering Play Mode before this PR shows a few errors. They're benign but look scary:
image

This PR would also make it easier to support editor mode in the future.

### CODE REVIEW

File: DisguiseUnityRenderStream/Runtime/NativeRenderingPlugin.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#if UNITY_STANDALONE_WIN && UNITY_64 && !UNITY_EDITOR
+#if UNITY_STANDALONE_WIN && UNITY_64
```
Comment (mark-beiline) [inline DisguiseUnityRenderStream/Runtime/NativeRenderingPlugin.cs:1]: The graphics init errors are coming from the `#if NATIVE_RENDERING_PLUGIN_AVAILABLE` (line 71 ) placeholder version of the NativeRenderingPluginNative class.

The solution is to not use the placeholder version in Play Mode.

File: DisguiseUnityRenderStream/Runtime/Util/AutoDisposable.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {
+#if UNITY_EDITOR
+            EditorApplication.quitting += Dispose;
+            AssemblyReloadEvents.beforeAssemblyReload += Dispose;
+#else
+            Application.quitting += Dispose;
+#endif
+        }
+
+        public virtual void Dispose()
```
Comment (wenzhang-unity) [inline DisguiseUnityRenderStream/Runtime/Util/AutoDisposable.cs]: This method can be made `abstract`. This class does not need to inherit `IDisposable`.

File: DisguiseUnityRenderStream/Runtime/TemporaryTexture2DManager.cs [hunk 3]
```diff
@@ -112,6 +112,19 @@ public static TemporaryTexture2DManager Instance
             PlayerLoopExtensions.RegisterUpdate<PostLateUpdate.FinishFrameRendering, FinishFrameRendering>(OnFinishFrameRendering);
         }
 
+        public override void Dispose()
+        {
+            foreach (var (_, item) in m_Items)
```
Comment (wenzhang-unity) [inline DisguiseUnityRenderStream/Runtime/TemporaryTexture2DManager.cs]: Curious about why we need this. Doesn't the runtime clean up the textures when the game quits, or when entering play mode (domain reload).

**Review by wenzhang-unity — CHANGES_REQUESTED** [review]

File: DisguiseUnityRenderStream/Runtime/TemporaryTexture2DManager.cs (hunk 3 above)
Comment (wenzhang-unity) [reply DisguiseUnityRenderStream/Runtime/TemporaryTexture2DManager.cs]: I have not encountered a need for this in any other projects I've seen.

File: DisguiseUnityRenderStream/Runtime/NativeRenderingPlugin.cs [hunk 4]
```diff
@@ -216,7 +216,7 @@ public EventDataPool()
             }
         }
         
-        public void Dispose()
+        protected override void Dispose()
```
Comment (wenzhang-unity) [inline DisguiseUnityRenderStream/Runtime/NativeRenderingPlugin.cs:219]: This a workaround for `NativeArray` complaining about `Dispose()` not being called when you perform a domain reload, right? It's super-annoying.

**Review by wenzhang-unity — APPROVED** [review]

File: DisguiseUnityRenderStream/Runtime/NativeRenderingPlugin.cs (hunk 4 above)
Comment (mark-beiline) [reply DisguiseUnityRenderStream/Runtime/NativeRenderingPlugin.cs:219]: Yes exactly

**Pull request merged**